sw/airborne/subsystems/datalink/bluegiga.c

```diff
@@ -0,0 +1,249 @@
+/*
+ * Copyright (C) 2014 Kirk Scheper <kirkscheper@gmail.com>
+ *
+ * This file is part of paparazzi.
+ *
+ * paparazzi is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2, or (at your option)
+ * any later version.
+ *
+ * paparazzi is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with paparazzi; see the file COPYING. If not, write to
+ * the Free Software Foundation, 59 Temple Place - Suite 330,
+ * Boston, MA 02111-1307, USA.
+ *
+ */
+
+/**
+ * @file subsystems/datalink/bluegiga.c
+ * Datalink implementation for the BlueGiga Bluetooth radio chip trough SPI
+ */
+
+#include "mcu_periph/sys_time.h"
+#include "subsystems/datalink/bluegiga.h"
+#include "mcu_periph/spi.h"
+#include "mcu_periph/gpio.h"
+
+#ifndef BLUEGIGA_SPI_DEV
+#error "bluegiga: must define a BLUEGIGA_SPI_DEV"
+#endif
+
+#ifndef BLUEGIGA_SLAVE_IDX
+#define BLUEGIGA_SLAVE_IDX SPI_SLAVE2
```

Comment:
This does not make any sense, if I see it correctly, then the bluegiga_spi.slave_idx is not even used anywhere in the slave code.
Better completely remove this and set slave_idx to zero below, so that it works if SPI_SLAVE2 is not defined